src/typesystem.rs

```diff
@@ -214,7 +214,7 @@ impl DataType {
             }
             TypeNature::Char => matches!(other.nature, TypeNature::Char | TypeNature::String),
             TypeNature::String => matches!(other.nature, TypeNature::String),
-            TypeNature::Any => true,
+            TypeNature::Any | TypeNature::Derived => true,
```

Comment:
I don't think this is correct since this will say all types are derived (including ints)